File: python/Problem392.py
# Problem: 392. Is Subsequence
# URL: https://leetcode.com/problems/is-subsequence/
# Difficulty: Easy
# Tags: Array, String, Subsequence

class Solution:
  def isSubsequence(self, s: str, t: str) -> bool:
    i: int = 0
    for j in range(len(t)):
      if i >= len(s):
        break
      if s[i] == t[j]:
        i += 1
    return i == len(s)

File: cpp/Problem392.cpp
/*
 * Problem: 392. Is Subsequence
 * URL: https://leetcode.com/problems/is-subsequence/
 * Difficulty: Easy
 * Tags: Array, String, Subsequence
 */

#include <string>

using namespace std;

class Solution {
public:
	bool isSubsequence(string s, string t) {
		int i = 0, j = 0;
		for (; i < s.length() && j < t.length(); j++) {
			if (s[i] == t[j]) {
				i++;
			}
		}
		return i == s.length();
	}
};